### PR TITLE
Feature/fastp and Remove dead code

### PR DESCRIPTION
**Pull request by matthdsm**
Added fastp qc step

## PR checklist
 - [x] This comment contains a description of changes (with reason)
 - [x] If you've fixed a bug or added code that should be tested, add tests!
 - [x] If necessary, also make a PR on the [nf-core/demultiplex branch on the nf-core/test-datasets repo]( https://github.com/nf-core/test-datasets/pull/new/nf-core/demultiplex)
 - [x] Ensure the test suite passes (`nextflow run . -profile test,docker`).
 - [x] Make sure your code lints (`nf-core lint .`).
 - [ ] Documentation in `docs` is updated
 - [ ] `CHANGELOG.md` is updated
 - [ ] `README.md` is updated

**Learn more about contributing:** https://github.com/nf-core/demultiplex/tree/master/.github/CONTRIBUTING.md

### CODE REVIEW

**Comment by matthdsm** [comment]
fixes #35

File: workflows/demultiplex.nf [hunk 1]
```diff
@@ -170,6 +164,41 @@ workflow.onComplete {
...
+    .groupTuple(by: [0])
+    // Add meta.single_end
+    .map {
+        meta, fastq ->
+        if (fastq.size() == 1){
+            meta.single_end = true
+        } else {
+            meta.single_end = false
+        }
+        return [ meta, fastq.flatten() ]
```
Comment (matthdsm) [inline workflows/demultiplex.nf:198]: Ideally, this should merge sample meta with `fc_meta`

**Comment by matthdsm** [comment]
@Emiller88 , can I get a ✅ ?

**Review by Emiller88 — CHANGES_REQUESTED** [review]
Awesome!

Let's just cut out the "legacy" modules, however. Those will be in the git history forever, so we don't have to keep them around. If we want to resurrect them, or pull parts out later, we can. I believe the scripts in `bin` can be removed as well.

You could also cherry-pick out those changes, we can merge this PR and then address those in a seperate PR.

**Review by Emiller88 — APPROVED** [review]
Awesome!

**Pull request merged**